submdspan_future/submdspan-future.md

```diff
@@ -987,71 +937,58 @@ template<class IndexType, size_t N, class... SliceSpecifiers>
 
-[*Note 2*: If $S_k$ is a valid $k^{th}$ `submdspan` slice type for `E`, then it is also a canonical $k^{th}$ `submdspan` slice type for `E`. -- *end note*]
+[5]{.pnum} Given an object `e` of type `E` that is specialization of `extents`,
+an object `s` of type `S`, **`s` is valid slice for `k` dimension of `e`** if:
 
-[6]{.pnum} Given an `extents` specialization `E`, an object `e` of type `E`, and a rank index $k$ of `E`, an object $s_k$ of type $S_k$ is a *valid $k^{th}$ `submdspan` slice for `e`* if $S_k$ is a valid $k^{th}$ `submdspan` slice type for `E`, and if all of the following are `true`:
+* [5.1]{.pnum} `S` is valid slice type for `k` dimension of `E`, and
 
-* [6.1]{.pnum} If $S_k$ is a specialization of `strided_slice`, then either
+* [5.2]{.pnum} interval of `k` dimension of `e` contains slice range of `s` for `k` dimension of `e`.
 
-    * [6.1.1]{.pnum} $s_k$`.extent` = 0, or
+[6]{.pnum} Given an object `e` of type `E` that is specialization of `extents`,
+ an pack of expression `slcs`, **`slcs` are valid slices for `e`** if:
 
-    * [6.1.2]{.pnum} $s_k$`.stride` &gt; 0; and
+* [6.1]{.pnum} `sizeof...(slcs)` is equal to `E::rank()`,
 
-* [6.2]{.pnum} 0 &le; _`first_`_`<IndexType, `$k$`>(slices...)` &le; _`last_`_`<`$k$`>(src, slices...)` &le; `src.extent(`$k$`)`.
+* [6.2]{.pnum} for each rank index `r` of `e`, the `slcs...[r]` is valid slice for `r` dimension of `e`.
 
-[7]{.pnum} Given an `extents` specialization `E`, an object `e` of type `E`, and a rank index $k$ of `E`, an object $s_k$ of type $S_k$ is a *canonical $k^{th}$ `submdspan` slice for `e`* if $s_k$ is a valid $k^{th}$ `submdspan` slice for `E`, and if $S_k$ is a canonical $k^{th}$ `submdspan` slice type for `E`.
 
 ```
 template<class IndexType, size_t... Extents, class... Slices>
 constexpr auto submdspan_canonicalize_slices(
-  const extents<IndexType, Extents...>& src, Slices... slices);
+  const extents<IndexType, Extents...>& src, Slices... raw_slices);
 ```
 
-[8]{.pnum} *Constraints*: `sizeof...(slices)` equals `sizeof...(Extents)`.
-
-[9]{.pnum} *Mandates*: For each rank index $k$ of `src`, $S_k$ is a valid $k^{th}$ `submdspan` slice type for `extents<IndexType, Extents...>`.
+Let `slices` denote pach _`canonical-slice`_`<IndexType>(raw_slices)`,
 
-[*Editorial Note*: This cannot be a Constraint because it is impossible to constrain a template on whether a structured binding declaration is well-formed.  This is because a compound requirement accepts an expression, not a declaration. -- *end note*]
+[8]{.pnum} *Constraints*: `sizeof...(raw_slices)` equals `sizeof...(Extents)`.
 
-[10]{.pnum} *Preconditions*: For each rank index $k$ of `src`, $s_k$ is a valid $k^{th}$ `submdspan` slice for `src`.
+[9]{.pnum} *Mandates*: For each rank index $k$ of `src`, $slices...[k]$ is valid expression and `decltype(slices...[k])` is a valid slice type for `k` dimension of `src`.

```

Comment:
The `canonical-slice` is valid expressions, brinds the Mandates about it being an acceptable slice type. I.e. the function is just ill-formed otherwise.